src/storage.ts

```diff
@@ -140,15 +129,12 @@ export class Storage {
         })
       return
     }
-    if (resp.status !== 200) {
```

Comment:
Could avoid piping to the temp file here by checking resp.status == 304 and returning the existing binary path.

Reply:
And also error if the status isn't 200 or 304 here so you don't have to pipe to the file first. I didn't think you could do that but you can judging by the similar check for 404 above.

Reply:
Good idea. I'll refactor.